tools/js-optimizer.js

```diff
@@ -6649,22 +6649,28 @@ function emterpretify(ast) {
             if (type === ASM_INT) {
               opcode = 'ADD';
               tryNumSymmetrical();
-            } else if (type === ASM_DOUBLE) opcode = 'ADDD';
+            } else if (type === ASM_DOUBLE || type === ASM_FLOAT) {
```

Comment:
Am personally -1 since there are only two float types, it's not that much extra noise - functions in the optimizer only tend to exist if there's some interesting logic in there that isn't immediately clear what it's doing (e.g. callHasSideEffects).

Also, I'll just have to move it out of the function anyway when I get to to porting the emterpreter to C++/Rust since I want it to switch across constants :)